### ISSUE TITLE
Information about vertical scroll of InputTextMultiline

### ISSUE THREAD

**Issue by tso996** [ocornut/imgui#7913]
### Version/Branch of Dear ImGui:

v1.89

### Back-ends:

imgui_impl_vulkan

### Compiler, OS:

Windows 11

I am trying to find the number of pixels scrolled up when we move the scroll bar down in an InputTextMultiline. ImGui seems to make the scrollbar when we add more newlines than can be contained in the InputTextMultiline widget if the height of the widget is fixed.
I was able to find the horizontal scroll information from a scrollbar I added to the same widget using the ImGuiInputTextState. Since the vertical scrollbar is made by ImGui itself, I'm not sure how to get the information.
Any suggestions are much appreciated.
Thanks!

**Comment by ocornut** [comment]
> I am trying to find the number of pixels scrolled up when we move the scroll bar down in an InputTextMultiline.

Why?
Answering the why almost always gives you a better answer.

You may enter the child window and query scrolling value:
```cpp
ImGui::InputTextMultiline("Editor", text, IM_ARRAYSIZE(text), ...);

ImGui::BeginChild("Editor");
float scroll_y = ImGui::GetScrollY();
float scroll_max_y = ImGui::GetScrollMaxY();
ImGui::EndChild();
ImGui::Text("%f/%f", scroll_y, scroll_max_y);
```

This is technically undocumented as it is possible we may attempt to remove the child window from `InputTextMultiline()` eventually (it's causing us a few issues).

I also pushed a small change as I feel it may make sense to expose it in our internal data structure, so if you update you may also do:

```cpp
#include "imgui_internal.h"
....
ImGui::InputTextMultiline("##source", text, IM_ARRAYSIZE(text), ImVec2(-FLT_MIN, ImGui::GetTextLineHeight() * 16), flags);
if (ImGuiInputTextState* state = ImGui::GetInputTextState(ImGui::GetItemID()))
    ImGui::Text("%f", state->Scroll.y);
```

**Comment by tso996** [comment]
@ocornut  Hey, thanks very much for responding so fast. As to why I need this info, I'm using ImGui and Qt together. That is, I'm running ImGui inside a Vulkan context in a Qt Window. I need to make changes to a QWidget which is external to the graphics context in which ImGui is running based on the contents visible in InputTextMultiline Widget. 
Updating ImGui is not an option at the moment as other components of the project will also need to be examined.
So I will try out your first suggestion.

**Comment by ocornut** [comment]
FYI there was a bug in this code where using `ImGuiInputTextFlags_NoHorizontalScroll` would mistakenly clear the exposed Scroll.y. This is fixed by e2f314d.
